src/main/java/net/minecraftforge/gametest/GameTestMain.java

```diff
@@ -0,0 +1,195 @@
+                LOGGER.warn("Failed to load datapacks, can't proceed with server load. You can either fix your datapacks or reset to vanilla with --safeMode", e);
+                packRepository.close();
+                return;
+            }
+
+            resources.updateGlobals();
+            RegistryReadOps<Tag> registryOps = RegistryReadOps.createAndLoad(NbtOps.INSTANCE, resources.getResourceManager(), registryHolder);
+            serverSettings.getProperties().getWorldGenSettings(registryHolder);
+            WorldData worldData = storageAccess.getDataTag(registryOps, datapacks);
+            if (worldData == null) {
+                DedicatedServerProperties dedicatedserverproperties = serverSettings.getProperties();
+                LevelSettings levelsettings = new LevelSettings(dedicatedserverproperties.levelName, dedicatedserverproperties.gamemode, dedicatedserverproperties.hardcore,
+                        dedicatedserverproperties.difficulty, false, new GameRules(), datapacks);
+                WorldGenSettings worldgensettings = dedicatedserverproperties.getWorldGenSettings(registryHolder);
+
+                // Forge: Deserialize the DimensionGeneratorSettings to ensure modded dims are loaded on first server load (see SimpleRegistryCodec#decode). Vanilla behaviour only loads from the
+                // server.properties and deserializes only after the 2nd server load.
+                worldgensettings = WorldGenSettings.CODEC
+                        .encodeStart(RegistryWriteOps.create(NbtOps.INSTANCE, registryHolder), worldgensettings)
+                        .flatMap(nbt -> WorldGenSettings.CODEC.parse(registryOps, nbt))
+                        .getOrThrow(false, errorMsg -> {});
+                worldData = new PrimaryLevelData(levelsettings, worldgensettings, Lifecycle.stable());
+            }
+
+            storageAccess.saveDataTag(registryHolder, worldData);
+            Collection<GameTestBatch> testBatches = GameTestRunner.groupTestsIntoBatches(GameTestRegistry.getAllTestFunctions());
+            BlockPos spawnPos = optionSet.valueOf(spawnPosOpt);
+            var server = MinecraftServer.spin(serverThread -> new GameTestServer(serverThread, storageAccess, packRepository, resources, testBatches, spawnPos, registryHolder));
+            Thread thread = new Thread("Server Shutdown Thread") {
+                @Override
+                public void run() {
+                    // We don't need to halt here as GameTestServer always calls System#exit on both crash and normal exit
+                    LogManager.shutdown(); // we're manually managing the logging shutdown on the server. Make sure we do it here at the end.
+                }
+            };
+            thread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler(LOGGER));
+            Runtime.getRuntime().addShutdownHook(thread);
+        } catch (Exception e) {
+            LOGGER.fatal("Failed to start the minecraft server", e);
```

Comment:
This logger message probably should mention something about game test minecraft server rather than just minecraft server.